### PR TITLE
Dry-run each query in CircleCI against prod datasets

### LINKED ISSUE #188: Dry run queries in CI

### PR DESCRIPTION
**Pull request by washort**
fixes #188

### CODE REVIEW

File: script/dryrun [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    try:
+        print(sql, end=pad)
+        r = urlopen(
+            Request(
+                DRY_RUN_URL,
+                headers = {
+                    'Content-Type': 'application/json'
+                },
+                data=json.dumps({
+                    'dataset': 'telemetry',
```
Comment (fbertsch) [inline script/dryrun]: Currently not every query runs on `telemetry`. Some run in `telemetry_live`. It depends on https://github.com/mozilla/bigquery-etl/issues/189 for us to know which dataset a query runs in.

File: script/dryrun [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import glob
+import json
+import os
+from urllib.request import urlopen, Request
+
+DRY_RUN_URL = 'https://us-central1-moz-fx-data-derived-datasets.cloudfunctions.net/bigquery-etl-dryrun'
+exitcode = 0
+for sql in glob.glob('sql/*.sql'):
```
Comment (relud) [inline script/dryrun]:
```suggestion
for sql in glob.glob('sql/*/*.sql'):
```

File: script/dryrun (hunk 1 above)
Comment (relud) [reply script/dryrun]: when #265 is merged this should produce the correct dataset:
```suggestion
                    'dataset': os.path.basename(os.path.dirname(sql)),
```

File: script/dryrun [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import glob
+import json
+import os
```
Comment (relud) [inline script/dryrun]:
```suggestion
import os.path
```

File: script/dryrun [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        exitcode = 1
+    else:
+        response = json.load(r)
+        if response['valid']:
+            print('OK')
+        else:
+            print('ERROR\n', response['errors'])
+            exitcode = 1
+
+os.exit(exitcode)
```
Comment (relud) [inline script/dryrun]:
```suggestion
sys.exit(exitcode)
```

File: script/dryrun [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import glob
+import json
+import os
+from urllib.request import urlopen, Request
```
Comment (relud) [inline script/dryrun]:
```suggestion
import sys
from urllib.request import urlopen, Request
```

File: script/dryrun [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import glob
+import json
+import os.path
+import sys
+from urllib.request import urlopen, Request
+
+DRY_RUN_URL = 'https://us-central1-moz-fx-data-derived-datasets.cloudfunctions.net/bigquery-etl-dryrun'
+exitcode = 0
+for sql in glob.glob('sql/*/*.sql'):
```
Comment (relud) [inline script/dryrun]:
```suggestion
skip={
    # Access Denied
    "sql/telemetry/fxa_content_events_v1.sql",
    "sql/telemetry/fenix_events_v1.sql",
    "sql/telemetry/fenix_events_amplitude_v1.sql",
    "sql/telemetry/fxa_auth_bounce_events_v1.sql",
    "sql/telemetry/fxa_auth_events_v1.sql",
    "sql/telemetry/glean_clients_daily_v1.sql",
    # Already exists (and lacks an "OR REPLACE" clause)
    "sql/telemetry/glean_clients_last_seen_raw_v1.init.sql",
    "sql/telemetry/core_clients_last_seen_raw_v1.init.sql",
    "sql/telemetry/fxa_users_last_seen_raw_v1.init.sql",
    # Reference table not found
    "sql/search/search_clients_daily_v7.sql",
    # Cannot access field list
    "sql/telemetry/lockwise_mobile_events_v1.sql",
    # No matching signature for function IF
    "sql/static/fxa_amplitude_export_users_last_seen.sql",
    # SQL functions with the same name but different signatures are currently unsupported
    "sql/static/fxa_amplitude_export_users_daily.sql",
    # Syntax error
    "sql/telemetry_derived/clients_last_seen_v1.init.sql",
    # Unrecognized name: submission_date_s3
    "sql/telemetry_derived/clients_daily_v7.sql",
    "sql/telemetry_derived/addons_v3.sql",
    "sql/telemetry_derived/addons_aggregates_v3.sql",
}
for sql in glob.glob('sql/*/*.sql'):
    if sql in skip:
        continue
```

**Pull request merged**